app/src/main/java/org/fossasia/openevent/adapters/LocationsListAdapter.java

```diff
@@ -105,18 +112,26 @@ public void moveItem(int fromPosition, int toPosition) {
         notifyItemMoved(fromPosition, toPosition);
     }
 
-    class Viewholder extends RecyclerView.ViewHolder {
+    public class Viewholder extends RecyclerView.ViewHolder implements View.OnClickListener {
         TextView name;
         TextView floor;
 
         public Viewholder(View itemView) {
             super(itemView);
             itemView.setClickable(true);
-
+            itemView.setOnClickListener(this);
             name = (TextView) itemView.findViewById(R.id.location_name);
             floor = (TextView) itemView.findViewById(R.id.location_floor);
 
         }
 
+        @Override
+        public void onClick(View view) {
+            int adapterPosition = getAdapterPosition();
+            Microlocation current = microlocations.get(adapterPosition);
+            String name = current.getName();
+            mClickHandler.onClick(name, this);
```

Comment:
As suggested by u in lower comments, i will make the class static and hence mClickHandler static also this will solve  adapter leaks...Any hint for preventing memory leak here?

I am accessing member variable to handle the onClick events.